administrator/components/com_joomgallery/helpers/config.php

```diff
@@ -383,6 +383,10 @@ public function __construct($id = null)
       $config = JTable::getInstance('joomgalleryconfig', 'Table');
       $config->load($this->_id);
 
+      JPluginHelper::importPlugin('content');
+      $_mainframe = JFactory::getApplication('administrator');
```

Comment:
JoomConfig::__construct is also called by the frontend, should the event also be triggered when loading the configuration in the frontend and for what is it usefull?

Reply:
JoomConfig::__construct loads the configurations from the db and offers them to the application during run with $this->_config. With a plugin using the 'onContentPrepareData' trigger in the JoomConfig::__construct method you are able to manipulate the configuration object. For example you could add another configuartion rule wich is not part of the #__joomgallery_config table without the need of changing the JoomGallery-Core.

Reply:
Please don't use an underscore as the first letter of a local variable. Just call it **mainframe** (please remember also the other occurrences where you changed ```$this->_mainframe``` to ```$_mainframe```).

Correct me if I am wrong, but the constructor is also called in the frontend application, so you should just call  
```
$mainframe = JFactory::getApplication();
```
to get the right application object, otherwise you will create a new administrator application object when calling the constructor in the frontend.

Why should I add another configuartion option in a plugin without changing the JoomGallery core which I can not use inside JoomGallery without changing the JoomGallery core?